File: JS/PGS_할인 행사.js
function solution(want, number, discount) {
    let answer = 0;
    let target = 0;
    let total_cnt = number.reduce((a,b) => a+b, 0);
    
    const lists = {};
    
    for(let i = 0 ; i < number.length ; i++){
        lists[want[i]] = number[i];
    }
    
    let copy_lists = Object.assign({}, lists);
    
    while(target < discount.length){
        Flag = true
        for(let i = target ; i < target + total_cnt ; i++){
            if(i >= discount.length){
                target = discount.length;
                Flag = false
                break
            }else if(!(Object.keys(copy_lists).includes(discount[i]))){
                target = i + 1
                Flag = false
                break
            }else if(Object.keys(copy_lists).includes(discount[i])){
                if(copy_lists[discount[i]] <= 0){
                    copy_lists = Object.assign({},lists)
                    target += 1
                    Flag = false
                    break
                }else{
                    copy_lists[discount[i]] -= 1
                }
            }
        }
        if(Flag){
            answer += 1
            target += 1
        }
        copy_lists = Object.assign({},lists)
    }
    
    return answer;
}

File: Python/PGS_할인 행사.py
def solution(want, number, discount):
    answer = 0
    target = 0
    total_cnt = sum(number)
    
    lists = dict()
    
    for i in range(len(number)):
        lists[want[i]] = number[i]
    copy_lists = lists.copy()
    
    while target < len(discount):
        for i in range(target, target+total_cnt):
            if i >= len(discount):
                target = len(discount)
                break
            elif discount[i] not in want:
                target = i + 1
                break
            elif discount[i] in want:
                if copy_lists[discount[i]] <= 0:
                    copy_lists = lists.copy()
                    target += 1
                    break
                else:
                    copy_lists[discount[i]] -= 1
                    print(copy_lists[discount[i]])
        else:
            answer += 1
            target += 1
        copy_lists = lists.copy()
                        
    return answer